src/auction/proxy.py

```diff
@@ -0,0 +1,32 @@
+import sys
```

Comment:
@nullun  @cusma  - as always, would highly appreciate you taking a quick look/sanity check over the contracts at src/auction folder. The corresponding unit test also demonstrates how they work in detail. I still need to add a few more error cases aside from the `happy-paths` covered but aside from that any feedback is welcome ;)